### PR TITLE
Complete

### PR DESCRIPTION
**Pull request by levialmog**
Ready for your review

### CODE REVIEW

File: cup_of_join.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+def cup_of_join(*args, sep='-'):
+    res = []
+
+    if args is None:
```
Comment (talbarfm) [inline cup_of_join.py]: image
Will args ever be None?

Comment (talbarfm) [reply cup_of_join.py]: Use Ipython to check things like this

File: cup_of_join.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        print(res)
+
+    for value in args:
```
Comment (talbarfm) [inline cup_of_join.py]: use nested list comprehention and zip for this

File: cup_of_join.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        res.extend(value)
+        res.append(sep)
+
+    res.pop()
+    print(res)
+
+
+cup_of_join([1, 2], [2, 3], [3, 4])
```
Comment (talbarfm) [inline cup_of_join.py]: https://www.freecodecamp.org/news/if-name-main-python-example/ 
Read this and fix

File: cup_of_join.py (hunk 2 above)
Comment (talbarfm) [reply cup_of_join.py]: Or itertools.chain

File: cup_of_join.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+def cup_of_join(*args, sep='-'):
+    res = []
```
Comment (talbarfm) [inline cup_of_join.py]: res? pick a better name?

File: cup_of_join.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+def cup_of_join(*args, sep='-'):
```
Comment (talbarfm) [inline cup_of_join.py]: Cant you say anything more specific about these *args? maybe a better name?

Comment (talbarfm) [inline cup_of_join.py]: Dont use abbreviations - separator not sep

File: cup_of_join.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+    if args is None:
+        print(res)
+
+    for value in args:
+        res.extend(value)
+        res.append(sep)
+
+    res.pop()
+    print(res)
```
Comment (talbarfm) [inline cup_of_join.py]: Use returns, dont coupe input output operations (like print and input) with code logic. Print it outside

**Review by talbarfm — CHANGES_REQUESTED** [review]

**Review by lovodkin93 — APPROVED** [review]
looks very good.

**Pull request merged**